enter.c

```diff
@@ -161,15 +161,55 @@ int enter(struct entry_settings *opts)
 		errx(1, "cannot create NICs when not in a network namespace");
 	}
 
+
+	char cgroup_path[BUFSIZ];
+	/*
+	 * If there is a cgroup specified we need to create bst.<pid> subcgroup to join
+	 * Otherwise if no cgroup is specified but limits are applied bst will error out
+	 */
+	if (opts->nactiveclimits != 0 || opts->cgroup_path != NULL) {
+		// If cgroup path is unspecified then attempt to derive from /proc/self/cgroup
+		if (opts->cgroup_path == NULL) {
+			FILE *selfcgroupfd = fopen("/proc/self/cgroup", "r");
+			if (selfcgroupfd == NULL) {
+				err(1, "unable to derive current cgroup hierarchy from /proc/self/cgroup");
+			}
+
+			char selfcgroup[BUFSIZ];
+
+			while (fgets(selfcgroup, BUFSIZ, selfcgroupfd) != NULL) {
+				if (strncmp(selfcgroup, "0::/", 3) == 0) {
+					// Remove newline character read by fgets
+					selfcgroup[strcspn(selfcgroup, "\n")] = '\0';
+					break;
+				}
+			}
+			fclose(selfcgroupfd);
+
+			// Form path from default cgroups path + current cgroup path with removed 0:/
```

Comment:
What happens if you don't find the 0::/? It looks like this just continues on with the last line of /proc/self/cgroup, and adding 3 to it, which is going to be very wrong